### ISSUE TITLE
Return to dock command doesn't seem to work all the time

### ISSUE THREAD

**Issue by ejl463** [andyb2000/HACS-husqvarna_automower_ble#27]
Hello, I have added this through HACS and everything seems to be working fine except the return to dock command.

When I run it, it seems like sometimes the mower will return but most of the time it continues mowing. Even if I pause it and click the return to dock, the mower has just continued mowing.
 Does this command function differently than I am expecting?

I have a Gardena sileno minimo if this helps.

Thanks

**Comment by andyb2000** [comment]
Hi,
This is unfortunately due to the poor communication over bluetooth to the mowers, if you watch the logs/status you'll see it drops often and the state goes into unknown.
I was thinking of maybe adding a command queue, so when you send a mow/dock command it will hold that until it gets a valid reply.

For now, press dock a few times and it should get there!

**Comment by ejl463** [comment]
Thanks for replying - 

I do see that it goes unreachable frequently, but I don't seem to have this issue with start or pause at all if the mower is in a reachable state  - only the return to dock seems to have the issue. Does this command behave the same as the others?

**Comment by andyb2000** [comment]
yes command should be the same. The difference is typically the mower is out and about to has poorer bluetooth signal whilst busy mowing.
I'll have to do some investigations to be sure it's not struggling in another way but when I saw this on my own month ago that was the case that it was just dropping more often due to it being out mowing and further from my ble proxy.

On the pause and dock, it shouldn't be any different to the app. Mine certainly when I choose dock will find the lead home wire and go home using that method as per the app. I've never tried pause then dock so there may be something wrong with the sequence there, but if so that's in the upstream library rather than the hacs addon which I'd need to check into.
